addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/XMLConnection.java

```diff
@@ -151,12 +161,55 @@ public String sendReceive(String message) throws IOException {
         }
     }
 
+    private Charset getResponseCharset(HttpURLConnection connection, Charset defaultCharset) {
+
+        // See https://stackoverflow.com/a/3934280/1906057
+
+        Charset charset = defaultCharset;
+
+        String contentType = connection.getContentType();
+        String[] values = contentType.split(";"); // values.length should be 2
+        String charsetName = "";
+
+        // Example:
+        // Content-Type:text/xml; charset="utf-8"
+
+        for (String value : values) {
+            value = value.trim();
+
+            if (value.toLowerCase().startsWith("charset=")) {
+                charsetName = value.substring("charset=".length() + 1, value.length() - 1);
+                break;
+            }
+        }
+
+        if (!StringUtils.isEmpty(charsetName)) {
+            try {
+                charset = Charset.forName(charsetName);
+            } catch (UnsupportedCharsetException | IllegalCharsetNameException e) {
+                logger.warn("The charset {} provided in the response {} is not supported", charsetName, contentType);
+            }
+        }
+
+        logger.trace("The charset {} will be used to parse the response", charset.name());
```

Comment:
This will give a NPE if we just logged on line 190, maybe just log charset and trust the `toString` implementation

Reply:
Hmm the charset would never be null. It is assigned a default charset on line regardless of line 188.
Either way I change the implementations, so it will be different (along with relying on toString on the logging).